source/ui/dataman/AMScanView.cpp

```diff
@@ -412,7 +469,30 @@ void AMScanView::onScanAdded(AMScan *scan)
 
 	modeBar_->showSpectra_->setEnabled(!sources.isEmpty());
 	modeBar_->showSpectra_->setVisible(!sources.isEmpty());
+
+	// Apply changes to views.
+
 	spectrumView_->setDataSources(sources);
+
+	QString xUnits = "";
+
+	if (scan && scan->rawData() && scan->rawData()->scanAxesCount() >= 1) {
+		xUnits = scan->rawData()->scanAxisAt(0).units;
+	}
+
+	foreach (AMScanViewInternal *scanView, views_) {
+		if (scanView && scanView->tools()) {
```

Comment:
A suggestion:
It might be nicer if we encapsulate this code block in `AMScanViewInternal` with a function name like `setUnits(QString units)`. Then the code in this function will be like:
```
if (scanView)
    scanView->setUnits(xUnits);
```

The user of `AMScanViewInternal` doesn't need to know how the the units if passed to the MPlot tools.